sdks/python/apache_beam/transforms/core.py

```diff
@@ -525,6 +525,44 @@ def __init__(
     self.param_id = 'WatermarkEstimatorProvider'
 
 
+class _BundleContextParam(_DoFnParam):
+  """Allows one to use a context manager to manage bundle-scoped parameters.
+
+  The context will be entered at the start of each bundle and exited at the
+  end, equivalent to the `start_bundle` and `finish_bundle` methods on a DoFn.
+
+  The object returned from `__enter__`, if any will be substituted for this
+  parameter in invocations.
```

Comment:
We should mention somewhere that the order of the context invocations is unspecified.